### DISCUSSION TITLE
require 'sequel/core' vs require 'sequel' in rodauth_demo.rb

### DISCUSSION THREAD

**Discussion by jf** [jeremyevans/rodauth#408] [General]
I see in https://github.com/jeremyevans/rodauth/blob/master/demo-site/rodauth_demo.rb that `require 'sequel/core'` is used. In my own usage of Sequel (it's been a while, admittedly), I don't remember require-ing `sequel/core`. Is there any reason to do specifically do `require 'sequel/core'` as opposed to `require 'sequel'`?

**Comment by janko** [comment]
When you `require "sequel/core"`, you don't load Sequel model code that Rodauth doesn't use, which means faster require and less memory. For comparison, requiring `sequel/core` loads 7.5k LOC, while requiring `sequel` loads 11.5k LOC.

**Reply by jf** [reply]
thank you!

**Reply by jf** [reply]
just one clarification here before I consider this closed: by models you would be referring to any classes that inherit from `Sequel::Model`, right? whereas if I were to only use datasets I would be fine sticking to `require sequel/core`?

**Reply by jeremyevans** [reply]
Correct.
